### PR TITLE
Extract UpdateMetadataForIndexCallable

### PR DESCRIPTION
**Pull request by rhyshort**
Extract UpdateMetadataForIndexCallable to be a standalone class.

### CODE REVIEW

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/UpdateMetadataForIndexCallable.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        this.lastSequence = lastSequence;
+        this.indexName = indexName;
+    }
+
+    @Override
+    public Void call(SQLDatabase database) throws QueryException {
+        ContentValues v = new ContentValues();
+        v.put("last_sequence", lastSequence);
+        int row = database.update(QueryImpl.INDEX_METADATA_TABLE_NAME,
+                v,
```
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/UpdateMetadataForIndexCallable.java]: Probably best to right align these arguments like before:
```
int row = database.update(QueryImpl.INDEX_METADATA_TABLE_NAME,		
                                              v,		
                                              " index_name = ? ",		
                                              new String[]{indexName});
```

Comment (rhyshort) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/UpdateMetadataForIndexCallable.java]: done in f7b83bb

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/UpdateMetadataForIndexCallable.java [hunk 2]
```diff
@@ -38,9 +38,9 @@ public Void call(SQLDatabase database) throws QueryException {
         ContentValues v = new ContentValues();
         v.put("last_sequence", lastSequence);
         int row = database.update(QueryImpl.INDEX_METADATA_TABLE_NAME,
-                v,
-                " index_name = ? ",
-                new String[]{indexName});
+                                  v,
+                      " index_name = ? ",
```
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/UpdateMetadataForIndexCallable.java]: `" index_name = ? ",` is still out of alignment.

Comment (rhyshort) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/callables/UpdateMetadataForIndexCallable.java]: fixed in 012b4cf

**Pull request merged**